workflows/crisprseq_screening.nf

```diff
@@ -233,16 +233,41 @@ workflow CRISPRSEQ_SCREENING {
         }.set { ch_counts }
     }
 
-    if(params.rra) {
-        Channel.fromPath(params.contrasts)
-            .splitCsv(header:true, sep:';' )
+    if (params.contrasts) {
+
+        ch_samplesheet
+            .map { meta, _fastq -> [meta.condition, meta] }
+            .groupTuple(by: 0) // Group by condition
+            .map { condition, metas -> [condition, metas.collect { it.id }]}
+            .set { ch_samplesheet_conditions }
+
+        Channel
+            .fromPath(params.contrasts)
+            .splitCsv(header:true, sep:';')
+            .map { line -> [
+                    id: "${line.treatment.replace(",", "_")}_vs_${line.reference.replace(",", "_")}",
+                    treatment: line.treatment.split(','),
+                    reference: line.reference.split(',')
+                ]
+            }
             .set { ch_contrasts }
-        counts = ch_contrasts.combine(ch_counts)
+
+        ch_contrasts
+            .combine(ch_samplesheet_conditions.collect(flat: false).map{ [it] })
+            .map { contrast_meta, all_conditions ->
+                def treatment_samples = all_conditions.find { it[0] in contrast_meta.treatment }  // Find samples for each condition
+                def reference_samples = all_conditions.find { it[0] in contrast_meta.reference }
+                return [ id: contrast_meta.id,  treatment: treatment_samples[1].join(","), reference: reference_samples[1].join(",") ]
+            }
+            .combine(ch_counts)
+            .set{ ch_contrasts_counts }
```

Comment:
Could we add comments with an example structure of these channels? (ch_samplesheet_conditions, ch_contrasts, ch_contrasts_counts) to make it easier to follow this code